File: backend/src/main/java/com/dmart/service/OrderService.java
package com.dmart.service;

import com.dmart.dto.request.CheckoutRequestDto;
import com.dmart.dto.request.UpdateOrderStatusRequestDto;
import com.dmart.dto.response.OrderResponseDto;
import com.dmart.entity.OrderStatus;

import java.util.List;

public interface OrderService {

    OrderResponseDto checkout(CheckoutRequestDto dto);

    List<OrderResponseDto> getMyOrders();

    OrderResponseDto getMyOrderById(Long orderId);

    OrderResponseDto cancelMyOrder(Long orderId);

    List<OrderResponseDto> getAllOrders(OrderStatus status);

    OrderResponseDto getOrderByIdForStaff(Long orderId);

    OrderResponseDto updateOrderStatus(Long orderId, UpdateOrderStatusRequestDto dto);
}

File: frontend/src/services/orderService.js
import { request } from './api';

export const orderService = {
  // Customer
  checkout: (dto = {}) => request('/orders/checkout', { method: 'POST', body: JSON.stringify(dto) }),
  getMyOrders: () => request('/orders'),
  getMyOrderById: (orderId) => request(`/orders/${orderId}`),
  cancelMyOrder: (orderId) => request(`/orders/${orderId}/cancel`, { method: 'PUT' }),

  // Staff
  getStaffOrders: (status) => {
    const q = status ? `?status=${status}` : '';
    return request(`/staff/orders${q}`);
  },
  getStaffOrderById: (orderId) => request(`/staff/orders/${orderId}`),
  updateOrderStatusByStaff: (orderId, status) => request(`/staff/orders/${orderId}/status`, { method: 'PUT', body: JSON.stringify({ status }) }),

  // Admin
  getAdminOrders: (status) => {
    const q = status ? `?status=${status}` : '';
    return request(`/admin/orders${q}`);
  },
  getAdminOrderById: (orderId) => request(`/admin/orders/${orderId}`),
  updateOrderStatusByAdmin: (orderId, status) => request(`/admin/orders/${orderId}/status`, { method: 'PUT', body: JSON.stringify({ status }) }),
};